travis/before_install.sh

```diff
@@ -51,51 +51,59 @@ for f in ${CHANGED_FORMULAE};do
   deps=$(brew deps -1 --include-build ${f})
   echo "${f} dependencies: ${deps}"
 
-  if [ "$(echo ${deps} | grep -c 'python')" != "0" ];then
-    echo "Installing and configuring Homebrew Python"
-    # Already installed, upgrade, if necessary
-    brew outdated python@2 || brew upgrade python@2
+  # Upgrade Python3 to the latest version, before installing Python2. Per the discussion here
+  # https://discourse.brew.sh/t/brew-install-python3-fails/1756/3
+  if [ "$(echo ${deps} | grep -c 'python3')" != "0" ];then
+    echo "Installing and configuring Homebrew Python3"
+      brew outdated || brew upgrade python
 
     # Set up Python .pth files
     # get python short version (major.minor)
-    PY_VER=$(${HOMEBREW_PREFIX}/bin/python2 -c "import sys;print('{0}.{1}'.format(sys.version_info[0],sys.version_info[1]).strip())")
+    PY_VER=$(${HOMEBREW_PREFIX}/bin/python3 -c "import sys;print('{0}.{1}'.format(sys.version_info[0],sys.version_info[1]).strip())")
```

Comment:
I think this should be python (without 3)

Reply:
The un-versioned python commands are linked to /usr/local/opt/python/libexec/bin, so we would either need to change the path, or leave the 3 suffix. the python command is simply symlinked to python3.